streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -267,25 +277,32 @@ public long approximateNumEntries() {
 
     @Override
     public void flush() {
+        validateStoreOpen();
         lock.writeLock().lock();
         try {
+            validateStoreOpen();
```

Comment:
Do we need to validate twice before grabbing the lock? Maybe only the validation at line 283 is sufficient.

Reply:
Matthias felt the same, and I also agree, but I'd like to do it in a follow-on PR. I was just quickly fixing a concurrency bug on the side, since I added synchronization to close expressly for the purpose of serializing closing the store wrt these methods.

https://github.com/apache/kafka/pull/8368/files/c51b89b829dbd9f306f9c94e085f978155cc1eb3#r399572942